src/engine/sparqlExpressions/ConvertToNumericExpression.cpp

```diff
@@ -8,9 +10,9 @@ namespace detail::to_numeric {
 // class that converts an input `int64_t`, `double` or `std::string`
 // to a numeric value `int64_t` or `double`
 template <typename T>
-class ToNumericImpl {
+requires std::integral<T> || std::floating_point<T> class ToNumericImpl {
```

Comment:
Okay, your implementation actually retuires `same_as<int64_t, T> || <same_as<double, T>` because of the structure of your `if constexpr` s. Maybe change this.